core/src/main/java/org/apache/accumulo/core/client/impl/ZookeeperLockChecker.java

```diff
@@ -30,13 +29,13 @@
   private final ZooCache zc;
   private final String root;
 
-  ZookeeperLockChecker(Instance instance) {
-    this(instance, new ZooCacheFactory());
+  ZookeeperLockChecker(ClientContext context) {
+    this(context, new ZooCacheFactory());
   }
 
-  ZookeeperLockChecker(Instance instance, ZooCacheFactory zcf) {
-    zc = zcf.getZooCache(instance.getZooKeepers(), instance.getZooKeepersSessionTimeOut());
-    this.root = ZooUtil.getRoot(instance) + Constants.ZTSERVERS;
+  ZookeeperLockChecker(ClientContext context, ZooCacheFactory zcf) {
+    zc = zcf.getZooCache(context.getZooKeepers(), context.getZooKeepersSessionTimeOut());
```

Comment:
This is not a suggestion for this PR, but this change made me think we could possibly have a method on ClientContext for getting a ZooCache.

Reply:
I agree with @keith-turner that a future change could include something on ClientContext for getting access to ZK, but I don't think it should just return ZooCache... because I've found that to be quite complicated for whatever has a ZooCache instance. We should try to create an abstraction for our use of ZK, and return the API for that abstraction, so that the caller doesn't need to know about the details of our caching layer over ZK.